src/main/java/Problem1/Permutation.java

```diff
@@ -5,7 +5,25 @@
 
 public class Permutation {
     public static List<List<Integer>> permute(int[] nums) {
-        // homework
-        return new ArrayList<>(); // place holder
+        List<List<Integer>> set = new ArrayList<>();
+        ArrayList<Integer> copy = new ArrayList<>();
+        helper(set, copy, nums);
+        return set;
     }
-}
+
+    private static void helper(List<List<Integer>> set, List<Integer> copy, int[] nums) {
+        if (copy.size() == nums.length) {
```

Comment:
try to put code in less level of condition. here do this:

```java
 if (copy.size() == nums.length) {
            set.add(new ArrayList<>(copy));
            return;
}
```

and remove {} from the rest of the code.